common/util/ziplain.cc

```diff
@@ -0,0 +1,265 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "common/util/ziplain.h"
+
+#include <endian.h>
+#include <string.h>
+#include <zlib.h>
+
+#include <cstdio>
+#include <memory>
+#include <vector>
+
+#include "absl/strings/string_view.h"
+
+namespace verible {
+
+ByteSource MemoryByteSource(absl::string_view input) {
+  auto is_called = std::make_shared<bool>(false);
+  return [is_called, input]() -> absl::string_view {
+    if (!*is_called) {
+      *is_called = true;
+      return input;
+    }
+    return {};
+  };
+}
+
+ByteSource FileByteSource(const char *filename) {
+  FILE *f = fopen(filename, "rb");
+  if (!f) return nullptr;
+  struct State {
+    State(FILE *f) : file(f) {}
+    ~State() { fclose(file); }
+    FILE *file;
+    char buffer[65536];
+  };
+  auto state = std::make_shared<State>(f);  // capture req. copy-constructable
+  return [state]() -> absl::string_view {
+    size_t r = fread(state->buffer, 1, sizeof(State::buffer), state->file);
+    return {state->buffer, r};
+  };
+}
+
+// Header writer, taking care of little-endianness writing of fields.
+namespace {
+class HeaderWriter {
+ public:
+  HeaderWriter(char *buffer) : begin_(buffer), pos_(buffer) {}
+
+  HeaderWriter &AddInt16(uint16_t value) {
+    uint16_t le = htole16(value);  // NOP on most platforms
+    memcpy(pos_, &le, 2);
+    pos_ += 2;
+    return *this;
+  }
+  HeaderWriter &AddInt32(uint32_t value) {
+    uint32_t le = htole32(value);  // NOP on most platforms
+    memcpy(pos_, &le, 4);
+    pos_ += 4;
+    return *this;
+  }
+  HeaderWriter &AddLiteral(absl::string_view str) {
+    memcpy(pos_, str.data(), str.size());
+    pos_ += str.size();
+    return *this;
+  }
+
+  bool Write(const ByteSink &out) {
+    return out({begin_, static_cast<size_t>(pos_ - begin_)});
+  }
+
+  void Append(std::vector<char> *buf) {
+    buf->insert(buf->end(), begin_, static_cast<const char *>(pos_));
+  }
+
+ private:
+  const char *begin_;
+  char *pos_;
+};
+}  // namespace
+
+struct Encoder::Impl {
+  static constexpr int16_t kPkZipVersion = 20;  // 2.0, pretty basic.
+  struct CompressResult {
+    uint32_t input_crc;
+    size_t input_size;
+    size_t output_size;
+  };
+
+  Impl(int compression_level, ByteSink out)
+      : compression_level_(compression_level < 0   ? 0
```

Comment:
If you add (empty) comments at th end of the line, hopefully the formatter will latch onto the right formatting
```c++
  compression_level_(compression_level < 0 ? 0  //
                      : compression_level > 9  ? 9 //
                                               : compression_level)
```

You can run the formatter locally with `.github/bin/run-clang-format.sh`

Reply:
Didn't help. I tried many things. But it puzzles me why CI and my local run of the script produce different results.

I'll try again later today.